File: Backend/src/main/kotlin/com/example/backend/data/NationalId.kt
package com.example.backend.data

import jakarta.persistence.*
import org.springframework.data.annotation.Id
import java.util.*

@Entity
@Table(name = "national_id")
data class NationalId(
    @jakarta.persistence.Id @GeneratedValue @Id
    var id: Long = 0,
    val name: String? = null,
    val sex: Char? = null,
    val nationality: String? = null,
    val dateOfBirth: Date? = null,
    val dateOfExpiry: Date? = null,
    val documentNr: String? = null,
    val can: String? = null,
    val placeOfBirth: String? = null,
    val nameAtBirth: String? = null,
    val mothersName: String? = null,
    val authority: String? = null,
)

File: ClipboardExtension/national-id.js
export class NationalId {
    constructor(data) {
        if (!data) {
            this.name = null;
            this.sex = null;
            this.nationality = null;
            this.dateOfBirth = null;
            this.dateOfExpiry = null;
            this.documentNr = null;
            this.can = null;
            this.placeOfBirth = null;
            this.nameAtBirth = null;
            this.mothersName = null;
            this.authority = null;
            return;
        }
        this.name = data.name;
        this.sex = data.sex;
        this.nationality = data.nationality;
        this.dateOfBirth = data.dateOfBirth;
        this.dateOfExpiry = data.dateOfExpiry;
        this.documentNr = data.documentNr;
        this.can = data.can;
        this.placeOfBirth = data.placeOfBirth;
        this.nameAtBirth = data.nameAtBirth;
        this.mothersName = data.mothersName;
        this.authority = data.authority;
    }
}